ingest/expression_files/mtx.py

```diff
@@ -75,6 +82,34 @@ def check_valid(
             raise ValueError("; ".join(error_messages))
         return True
 
+    @staticmethod
+    def is_sorted(file_path: str):
+        """Checks if a file is sorted by gene index"""
+        file_size = os.path.getsize(file_path)
+        if file_size == 0:
+            raise ValueError(f"{file_path} is empty: " + str(file_size))
+        start_idx: int = MTXIngestor.get_data_start_line_number(file_path)
+        # Grab gene expression data which starts at 'n', or start_idx, lines from top of file (-n +{start_idx}).
+        # The header and mtx dimension aren't included or else the file would always be considered file due to the mtx
+        # dimensions always being larger than the first row of data.
+        p1 = subprocess.Popen(
+            ["tail", "-n", f"+{start_idx}", f"{file_path}"], stdout=subprocess.PIPE
+        )
+        # Check that the input file is sorted (-c). Check sorting by first and only first column (-k 1,1,). This value is
+        # numeric(-n). Use stable sort (--stable) so that columns are only compared by the first column.
+        # Without this argument line '1 3 4' and ' 1 5 4' would be considered unsorted.
+        p2 = subprocess.run(
+            ["sort", "-c", "--stable", "-n", "-k", "1,1"],
```

Comment:
A recall discussion of a particular `sort` version being important.  

To avoid difficult re-discovery of that tricky bit, could add a comment on which version of `sort` is required?

Reply:
8d59336